lib/Dialect/AIEVec/Transforms/AIEVectorize.cpp

```diff
@@ -1954,6 +1954,248 @@ static void fuseFMAOpsForColumnTopology(func::FuncOp func, VectState *state) {
     op->erase();
 }
 
+// We go through each fma operation and try to find the patten like this-
+// the acc of fma is a mul/fma operation which uses the same operands as fma.
+// the def of two operands are upd operations.
+// Transform -
+// %5 = aievec.mul %4, %0 {xoffsets = "0x03020100", xoffsets_hi = "0x07060504",
+// xsquare = "0x2110", xstart = "0", zoffsets = "0", zoffsets_hi = "0", zstart =
+// "0", zstep = "1"}
+//
+// %6 = aievec.mac %4, %0, %5 {xoffsets = "0x03020100",
+// xoffsets_hi = "0x07060504", xsquare = "0x2110", xstart = "2", zoffsets = "0",
+// zoffsets_hi = "0", zstart = "2", zstep = "1"}
+//
+// to-
```

Comment:
Replace the values of xoffsets, xoffsets_hi, xsquare, with symbolic constants. This would make it easier to spot which attribute among %5 and %6 that need to have the same value, and which attributes need to differ.